hardware_isolation_manager.cpp

```diff
@@ -326,4 +327,105 @@ std::optional<sdbusplus::message::object_path>
+{
+    auto entityPathRawData =
+        devtree::convertEntityPathIntoRawData(record.targetId);
+    std::stringstream ss;
+    std::for_each(entityPathRawData.begin(), entityPathRawData.end(),
+                  [&ss](const auto& ele) {
+                      ss << std::setw(2) << std::setfill('0') << std::hex
+                         << (int)ele << " ";
+                  });
+
+    try
+    {
+        entry::EntryResolved resolved = false;
+        if (record.recordId == 0xFFFFFFFF)
+        {
+            resolved = true;
+        }
+
+        auto isolatedHwInventoryPath =
+            _isolatableHWs.getInventoryPath(entityPathRawData);
+
+        if (!isolatedHwInventoryPath.has_value())
+        {
+            log<level::ERR>(
+                fmt::format(
+                    "Skipping to restore a given isolated "
+                    "hardware [{}] : Due to failure to get inventory path",
+                    ss.str())
+                    .c_str());
+            return;
+        }
+
+        auto bmcErrorLogPath = getBMCLogPath(record.elogId);
+
+        if (!bmcErrorLogPath.has_value())
+        {
+            log<level::ERR>(
+                fmt::format(
+                    "Skipping to restore a given isolated "
+                    "hardware [{}] : Due to failure to get BMC error log path "
+                    "by isolated hardware EID (aka PEL ID) [{}]",
+                    ss.str(), record.elogId)
+                    .c_str());
+            return;
+        }
+
+        auto entrySeverity = entry::utils::getEntrySeverityType(
+            static_cast<openpower_guard::GardType>(record.errType));
+        if (!entrySeverity.has_value())
+        {
+            log<level::ERR>(
+                fmt::format("Skipping to restore a given isolated "
+                            "hardware [{}] : Due to failure to to get BMC "
+                            "EntrySeverity by isolated hardware GardType [{}]",
+                            ss.str(), record.errType)
+                    .c_str());
+            return;
+        }
+
+        auto entryPath = createEntry(record.recordId, resolved, *entrySeverity,
+                                     isolatedHwInventoryPath->str,
+                                     bmcErrorLogPath->str, false);
+
+        if (!entryPath.has_value())
+        {
+            log<level::ERR>(
+                fmt::format(
+                    "Skipping to restore a given isolated "
+                    "hardware [{}] : Due to failure to create dbus entry",
+                    ss.str())
+                    .c_str());
+            return;
+        }
+    }
+    catch (const std::exception& e)
+    {
+        log<level::ERR>(
+            fmt::format("Exception [{}] : Skipping to restore a given isolated "
+                        "hardware [{}]",
+                        e.what(), ss.str())
+                .c_str());
+    }
+}
+
+void Manager::restore()
```

Comment:
General comment : restore is going to call at begining of Guard app. You need to make sure guard partion is created. Also incase you are explicitly checking the guard file path , need to be carefull , in genesis mode rw creates only when fisrt attempt of gurad partion. ( not sure any chnage here related tohis recent)  . Need to handle this corectly ,. iremebre huard lib already handles this , incase app explictly checks need to be careful. Adding this command here , to just make sure need to consider these points.